### PR TITLE
neovim-qt: move to pkgs/by-name

### PR DESCRIPTION
**Pull request by teto**
## Description of changes

2nd attempt of https://github.com/NixOS/nixpkgs/pull/309074

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by teto** [comment]
Result of `nixpkgs-review pr 316408` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>neovim-qt</li>
  </ul>
</details>

File: pkgs/by-name/ne/neovim-qt-unwrapped/package.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
-{ lib, mkDerivation, fetchFromGitHub, cmake, doxygen, msgpack, neovim, python3Packages, qtbase, qtsvg }:
+{ lib, libsForQt5, fetchFromGitHub, cmake, doxygen, msgpack, neovim, python3Packages }:
 
-mkDerivation rec {
+libsForQt5.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/by-name/ne/neovim-qt-unwrapped/package.nix:3]: Use stdenv.mkDerivation instead.

File: pkgs/by-name/ne/neovim-qt-unwrapped/package.nix [hunk 2]
```diff
@@ -16,7 +16,7 @@ mkDerivation rec {
     "-DENABLE_TESTS=0"  # tests fail because xcb platform plugin is not found
   ];
 
-  buildInputs = [
+  buildInputs = with libsForQt5; [
```
Comment (AndersonTorres) [inline pkgs/by-name/ne/neovim-qt-unwrapped/package.nix:19]: Never use `with` this way. Use `libsForQt5.qtXYZ` directly.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ne/neovim-qt-unwrapped/package.nix (hunk 1 above)
Comment (teto) [reply pkgs/by-name/ne/neovim-qt-unwrapped/package.nix:3]: I mimicked what was done at pkgs/by-name/qd/qdirstat/package.nix

Comment (AndersonTorres) [reply pkgs/by-name/ne/neovim-qt-unwrapped/package.nix:3]: The message points to a wrapQtAppsHook missing at nativeBuildInputs.

Comment (teto) [reply pkgs/by-name/ne/neovim-qt-unwrapped/package.nix]: is this better ?

**Comment by teto** [comment]
@AndersonTorres gentle ping :) 

NB to potential merger: Dont forget to squash before merge

**Comment by AndersonTorres** [comment]
> 
> NB to potential merger: Dont forget to squash before merge

According to the guidelines, this is part of your job.

**Review by AndersonTorres — APPROVED** [review]

**Comment by teto** [comment]
thank you !

**Pull request merged**